File: src/main/java/cn/dsc/ufocus/param/user/UserRoleRel.java
package cn.dsc.ufocus.param.user;

import cn.dsc.ufocus.param.IKeyBase;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author ding.shichen
 */
@EqualsAndHashCode
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserRoleRel implements IKeyBase<Long> {

    /**
     * 用户ID
     */
    private Long userId;

    /**
     * 角色ID
     */
    private Long roleId;

    @Override
    public Long getKey() {
        return userId;
    }

    @Override
    public void setKey(Long key) {
        this.userId = key;
    }
}

File: src/main/kotlin/cn/dsc/ufocus/convert/UserRoleRel.kt
// @author  ding.shichen
// @email   foreverhuiqiao@126.com
// @date    2023-12-11

package cn.dsc.ufocus.convert

import cn.dsc.ufocus.entity.UserRoleRelEntity
import cn.dsc.ufocus.param.role.RoleOption
import cn.dsc.ufocus.param.user.UserRoleRel

fun UserRoleRelEntity.toDetail() = UserRoleRel().also {
    it.userId = userId
    it.roleId = roleId
}

fun UserRoleRel.toRole() = RoleOption().also {
    it.id = roleId
}